mythril/laser/ethereum/transaction/transaction_models.py

```diff
@@ -44,7 +44,7 @@ def __init__(self,
         self.world_state = world_state
         self.callee_account = callee_account
         self.caller = caller
-        self.call_data = call_data
+        self.call_data = Calldata(self.id) if call_data is None else Calldata(self.id, call_data) if type(call_data) == list else call_data
```

Comment:
I feel like this might be getting a little bit too long of a one liner, wdyt?
Why should a list and Calldata object be admissible? Why not only Calldata?

Reply:
I'll shorten it to:
```
self.call_data = Calldata(self.id) or Calldata(self.id, call_data) if type(call_data) == list else call_data
```

If calldata is passed in as a list, I assume the concrete calldata bytes are being passed in.

Reply:
I assume that a list passed in is the concrete calldata.

I think it's also long, but a bunch of the one liners there are also really long. It's only a few characters longer than the next longest.
If someone is going to fix pep8, I think they can probably deal with it in conjunction with all of the other lines following it, and if not, the line fits in pretty well with all the other ternary lines following it.

Reply:
I kind of disagree, although the following lines could be shortened, the logic is quite simple
But this line has a nested if statement, and isn't simple anymore.
I'd either rewrite it as a normal if statement or limit the input to __init__ to either None or a list

Reply:
```
self.call_data = Calldata(self.id, call_data) if not isinstance(call_data, Calldata) else call_data
```
This seems nicer, I'll do this.